lib/ansible/plugins/lookup/aws_ssm.py

```diff
@@ -45,130 +48,150 @@
-    - 'region=ap-southeast-2'
-    - 'bypath'
+    - '/TEST/test-list region=ap-southeast-2, bypath'
 '''
+# FIXME the last one is probably not true yet. 
 
-from ansible.module_utils.ec2 import HAS_BOTO3
+from ansible.utils.display import Display
+from ansible.module_utils.ec2 import HAS_BOTO3, boto3_tag_list_to_ansible_dict
 from ansible.errors import AnsibleError
 from ansible.plugins.lookup import LookupBase
 from ansible.module_utils.parsing.convert_bool import boolean
+import pdb
+
+from ansible.utils.display import Display
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
 
 try:
     from botocore.exceptions import ClientError
+    import botocore
     import boto3
 except ImportError:
     pass  # will be captured by imported HAS_BOTO3
 
 
+def _boto3_conn(region, credentials):
+    if 'boto_profile' in credentials:
+        boto_profile = credentials.pop('boto_profile')
+    else:
+        boto_profile = None
+
+    try:
+        connection = boto3.session.Session(profile_name=boto_profile).client('ssm', region, **credentials)
+    except (botocore.exceptions.ProfileNotFound, botocore.exceptions.PartialCredentialsError) as e:
+        if boto_profile:
+            try:
+                connection = boto3.session.Session(profile_name=boto_profile).client('ssm', region)
+            except (botocore.exceptions.ProfileNotFound, botocore.exceptions.PartialCredentialsError) as e:
+                raise AnsibleError("Insufficient credentials found.")
+        else:
+            raise AnsibleError("Insufficient credentials found.")
+    return connection
+
+
 class LookupModule(LookupBase):
-    def run(self, terms, variables, **kwargs):
+    def run(self, terms, variables, boto_profile=None, aws_secret_key=None, aws_access_key=None,
+            aws_security_token=None, region=None, bypath=False, shortnames=False, recursive=False,
+            decrypt=True):
         '''
-            :param terms: a list of plugin options
-                          e.g. ['parameter_name', 'region=us-east-1', 'aws_profile=profile', 'decrypt=false']
-            :param variables: config variables
-            :return The value of the SSM parameter or None
+            :param terms: a list of lookups to run
+                e.g. ['parameter_name', 'parameter_name_too' ]
+            :param variables: ansible variables active at the time of the lookup
+            :keyword args: keyword arguments used for configuring the lookup
+                - aws_secret_key / aws_access_key / aws_security_token - AWS credentials
+                - region - the AWS region in which to do the lookup
+                - bypath - set to true to do a lookup of variables under a path
+                - recursive - set to true to do a recurse to paths below the path (needs bypath)
+            :return A list of parameter values or a list of dictionaries if doing a bypath lookup
```

Comment:
We have been using sphinx format for docstrings.  In sphinx format, this would be:
```
:arg terms: a list of lookups to run.
    e.g. ['parameter_name', 'parameter_name_too' ]
:kwarg variables: ansible variables active at the time of the lookup
:kwarg aws_secret_key: One part of AWS credentials
:kwarg aws_access_key: The second part of AWS credentials
:kwarg aws_security_token: Third part of the AWS credentials
:kwarg region: AWS region in which to do the lookup
:kwarg bypath: Set to True to do a lookup of variables under a path
:kwarg recursive: Set to True to recurse into paths below  the path (requires bypath=True)
:returns: A list of parameter values or a list of dictionaries if bypath=True.
```